plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -491,11 +489,17 @@ public static function verify_base_tables( $modify_notice = true, $execute = fal
 
 		$schema = self::get_schema();
 
-		$feature_controller = wc_get_container()->get( FeaturesController::class );
-		if (
-			$feature_controller->feature_is_enabled( DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION )
-			|| $feature_controller->feature_is_enabled( CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION )
-		) {
+		$hpos_settings = filter_var_array(
+			array(
+				'cot'       => get_option( CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION ),
+				'data_sync' => get_option( DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION ),
+			),
+			array(
+				'cot'       => FILTER_VALIDATE_BOOL,
+				'data_sync' => FILTER_VALIDATE_BOOL,
+			)
+		);
+		if ( in_array( true, $hpos_settings, true ) ) {
```

Comment:
Is this check really necessary now that HPOS is non-experimental and enabled by default on new installs? The tables should always exist, right?

Reply:
🤔 I would guess that it's an extremely unlikely scenario, but possible, that this check might still be necessary because the install routine would run again if you disabled and then re-enabled the plugin.